lib/kb_Metrics/Util.py

```diff
@@ -30,10 +31,15 @@ def _convert_to_datetime(dt):
     else:
         raise ValueError('Cannot convert {} to datetime'.format(dt))
 
+
 def _partition_by_keys(src_list, idKeys, dataKeys):
     idList = []
     dataList = []
+
     for src in src_list:
-        idList.append({id_k: src[id_k]}) for id_k in idKeys
-        dataList.append({data_k: src[data_k]}) for data_k in dataKeys
+        for id_k in idKeys:
```

Comment:
I don't quite understand what you are trying to do. 

Say you have id_k = ['a']
So idList will look like [{'a': 'some data form src_1'}, {'a': 'some data from src_2'}, {'a': 'some data from src_3'}]? How are you planning to identify which data is from src_1, src_2 or src_3?

Reply:
I had the lambda inline that worked well. But when pycodestyle recommended to use def, I thought of giving it a try.  It is not used/tested with data set as of now, so I can remove it.